udtube/callbacks.py

```diff
@@ -101,13 +102,22 @@ def _fill_in_tags(
             attr (str): attribute on tokens where the tags should be inserted.
             tags (Iterator[str]): tags to insert.
         """
-        # Note that in when MWEs are present, the iterators with predicted tags
+        # Note that when MWEs are present, the iterators with predicted tags
         # from the classifier heads are shorter than the tokenlists, so we
         # `continue` without advancing said iterators.
         for token in tokenlist:
             if token.is_mwe:
                 continue
-            setattr(token, attr, next(tags))
+            try:
+                setattr(token, attr, next(tags))
+            except StopIteration:
+                # Prevents the error from being caught by Lightning.
+                logging.error(
+                    "Length mismatch between the tags and the "
+                    f"tokenlist for tags ({attr}).\n"
+                    f"Tokenlist sent id: {tokenlist.metadata.get('sent_id')}."
```

Comment:
Sorry to bikeshed on this but log statements with newlines violates every expectation I have about logging. I think it would be fine to say something like "Length mismatch for tag {attr!r} (sent_id: ...)". 

`sent_id` can be absent. Are you okay with it putting `None` here? (I don't see any problem, just drawing attention to this.)